src/github/SyftGithubAction.ts

```diff
@@ -0,0 +1,398 @@
+        image: core.getInput("image"),
+      },
+      format: getSbomFormat(),
+    });
+
+    core.info(`SBOM scan completed in: ${(Date.now() - start) / 1000}s`);
+
+    if (output) {
+      const { eventName, payload, repo } = github.context;
+      if (comparePulls && eventName === "pull_request") {
+        const client = getClient(repo, core.getInput("github_token"));
+
+        const pr = (payload as PullRequestEvent).pull_request;
+        const branchWorkflow = await client.findLatestWorkflowRunForBranch({
+          branch: pr.base.ref,
+        });
+
+        core.debug("Got branchWorkflow");
+        core.debug(JSON.stringify(branchWorkflow));
+
+        if (branchWorkflow) {
+          const baseBranchArtifacts = await client.listWorkflowRunArtifacts({
+            runId: branchWorkflow.id,
+          });
+
+          core.debug("Got baseBranchArtifacts");
+          core.debug(JSON.stringify(baseBranchArtifacts));
+
+          for (const artifact of baseBranchArtifacts) {
+            if (artifact.name === getArtifactName()) {
+              const baseArtifact = await client.downloadWorkflowRunArtifact({
+                artifactId: artifact.id,
+              });
+
+              core.info(
+                `Downloaded SBOM from ${pr.base.ref} to ${baseArtifact}`
+              );
+            }
+          }
+        }
+      }
+
+      const priorArtifact = process.env[PRIOR_ARTIFACT_ENV_VAR];
+      if (priorArtifact) {
+        core.info(`Prior artifact: ${priorArtifact}`);
+      }
+
+      if (doUpload) {
+        await uploadSbomArtifact(output);
+      }
+
+      core.exportVariable(PRIOR_ARTIFACT_ENV_VAR, getArtifactName());
+
+      if (outputVariable) {
+        // need to escape multiline strings a specific way:
+        // https://github.community/t/set-output-truncates-multiline-strings/16852/5
+        const content = output
+          .replace("%", "%25")
+          .replace("\n", "%0A")
+          .replace("\r", "%0D");
+        core.setOutput(outputVariable, content);
+      }
+    } else {
+      core.error(JSON.stringify(output));
+    }
+  } catch (e) {
+    if (e instanceof Error) {
+      core.setFailed(e.message);
+    } else if (e instanceof Object) {
+      core.setFailed(JSON.stringify(e));
+    } else {
+      core.setFailed(`An unknown error occurred: ${e}`);
+    }
+    throw e;
+  }
+}
+
+/**
+ * Attaches the SBOM assets to a release if run in release mode
+ */
+export async function attachReleaseArtifacts(): Promise<void> {
+  const doRelease = getBooleanInput("upload_release_assets", true);
+
+  if (!doRelease) {
+    return;
+  }
+
+  try {
+    core.debug(dashWrap(`Got github context:`));
+    core.debug(JSON.stringify(github.context));
+
+    const { eventName, ref, payload, repo } = github.context;
+    const client = getClient(repo, core.getInput("github_token"));
+
+    let release: Release | undefined = undefined;
+
+    // FIXME: what's the right way to detect a release?
```

Comment:
This has a FIXME - Did you find that it was not falling into this block based on the `eventName` check?

Reply:
I still don't know this is necessarily correct. It seems to be working fine but I need to revisit.